repos/fdbt-site/src/pages/api/managePassengerTypes.ts

```diff
@@ -0,0 +1,216 @@
+const nameRequiredErrorMessage = 'You must provide a name';
+const nameGreaterThanMaxErrorMessage = 'The name caannot be greater than 50 characters';
+const ageRangeNumberError = 'Enter a whole number between 0-150';
+const ageRangeInputError = 'Enter a minimum or maximum age';
+const proofSelectionError = 'Select at least one proof document';
+const maxLessThanMinError = (inputType: string): string =>
+    `Maximum ${inputType} cannot be less than minimum ${inputType}`;
+const minGreaterThanMaxError = (inputType: string): string =>
+    `Minimum ${inputType} cannot be greater than maximum ${inputType}`;
+
+const minAgeRangeSchema = yup
+    .number()
+    .typeError(ageRangeNumberError)
+    .integer(ageRangeNumberError)
+    .min(0, ageRangeNumberError);
+
+const maxAgeRangeSchema = yup
+    .number()
+    .typeError(ageRangeNumberError)
+    .integer(ageRangeNumberError)
+    .max(150, ageRangeNumberError);
+
+export const requestValidationRules = yup
+    .object({
+        type: yup.string().required(passengerTypeRequiredErrorMessage),
+        name: yup.string().required(nameRequiredErrorMessage).max(50, nameGreaterThanMaxErrorMessage),
+        ageRangeMin: yup.number().when('ageRange', {
+            is: 'Yes',
+            then: yup
+                .number()
+                .when('ageRangeMax', {
+                    is: (ageRangeMaxValue) => !!ageRangeMaxValue,
+                    then: minAgeRangeSchema.max(yup.ref('ageRangeMax'), minGreaterThanMaxError('age')).notRequired(),
+                })
+                .when('ageRangeMax', {
+                    is: (ageRangeMaxValue) => !ageRangeMaxValue,
+                    then: minAgeRangeSchema.max(150, ageRangeNumberError).required(ageRangeInputError),
+                }),
+        }),
+        ageRangeMax: yup.number().when('ageRange', {
+            is: 'Yes',
+            then: yup
+                .number()
+                .when('ageRangeMin', {
+                    is: (ageRangeMinValue) => !!ageRangeMinValue,
+                    then: maxAgeRangeSchema.min(yup.ref('ageRangeMin'), maxLessThanMinError('age')).notRequired(),
+                })
+                .when('ageRangeMin', {
+                    is: (ageRangeMinValue) => !ageRangeMinValue,
+                    then: maxAgeRangeSchema.min(0, ageRangeNumberError).required(ageRangeInputError),
+                }),
+        }),
+        proofDocuments: yup.string().when('proof', { is: 'Yes', then: yup.string().required(proofSelectionError) }),
+    })
+    .required();
+
+export const formatRequestBody = (req: NextApiRequestWithSession): FilteredRequestBody => {
+    const filteredReqBody: IFilteredRequestBody = {};
+
+    Object.entries(req.body).forEach((entry) => {
+        const key = entry[0];
+        const value = entry[1];
+
+        if (key === 'type') {
+            filteredReqBody[key] = value as string;
+        }
+
+        if (key === 'name') {
```

Comment:
could this and the previous assertion be comined to `if (key === 'type || key ==='name') {...}`